contracts/validator/contract/XDCValidatorV2.sol

```diff
@@ -0,0 +1,521 @@
+pragma solidity ^0.4.26;
+
+import "./libs/SafeMath.sol";
+
+contract XDCValidator {
+    using SafeMath for uint256;
+
+    event Vote(address _voter, address _candidate, uint256 _cap);
+    event Unvote(address _voter, address _candidate, uint256 _cap);
+    event Propose(address _owner, address _candidate, uint256 _cap);
+    event Resign(address _owner, address _candidate);
+    event Withdraw(address _owner, uint256 _blockNumber, uint256 _cap);
+    event UploadedKYC(address _owner, string kycHash);
+    event InvalidatedNode(address _masternodeOwner, address[] _masternodes);
+
+    struct ValidatorState {
+        address owner;
+        bool isCandidate;
+        uint256 cap;
+        mapping(address => uint256) voters;
+    }
+
+    struct WithdrawState {
+        mapping(uint256 => uint256) caps;
+        uint256[] blockNumbers;
+    }
+
+    mapping(address => WithdrawState) withdrawsState;
+
+    mapping(address => ValidatorState) validatorsState;
+    mapping(address => address[]) voters;
+
+    // Mapping structures added for KYC feature.
+    mapping(address => string[]) public KYCString;
+    mapping(address => uint) public invalidKYCCount;
+    mapping(address => mapping(address => bool)) public hasVotedInvalid;
+    mapping(address => address[]) public ownerToCandidate;
+    address[] public owners;
+
+    address[] public candidates;
+
+    uint256 public candidateCount = 0;
+    uint256 public ownerCount = 0;
+    uint256 public minCandidateCap;
+    uint256 public minVoterCap;
+    uint256 public maxValidatorNumber;
+    uint256 public candidateWithdrawDelay;
+    uint256 public voterWithdrawDelay;
+    // owner => invalid
+    mapping(address => bool) public invalidOwner;
+    // candaite => invalid
+    mapping(address => bool) public invalidCandidate;
+
+    struct PendingKYCState {
+        uint256 blockNumber;
+        string kycHash;
+    }
+
+    mapping(address => PendingKYCState) public pendingKYC;
+
+    modifier onlyValidCandidateCap() {
+        // anyone can deposit X XDC to become a candidate
+        require(msg.value >= minCandidateCap, "Invalid Candidate Cap");
+        _;
+    }
+
+    modifier onlyValidVoterCap() {
+        require(msg.value >= minVoterCap, "Invalid Voter Cap");
+        _;
+    }
+
+    modifier onlyKYCWhitelisted() {
+        require(
+            KYCString[msg.sender].length != 0 ||
+                ownerToCandidate[msg.sender].length > 0,
+            "KYC not uploaded"
+        );
+        _;
+    }
+
+    modifier onlyOwner(address _candidate) {
+        require(
+            validatorsState[_candidate].owner == msg.sender,
+            "Only owner can call this function"
```

Comment:
"Only owner can call this function" is more than 32 bytes. The message should less than 32 bytes.